LargeStreamletTest

```diff
@@ -0,0 +1,160 @@
+        connection.start();
+        try {
+            final Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+            try {
+                final Destination destination = session.createQueue("wibble");
+                final Thread readerThread = new Thread(new Runnable() {
+
+                    public void run() {
+                        totalRead.set(0);
+                        try {
+                            final InputStream inputStream = connection.createInputStream(destination);
+                            try {
+                                int read;
+                                final byte[] buf = new byte[BUFFER_SIZE];
+                                while (!stopThreads.get() && (read = inputStream.read(buf)) != -1) {
+                                    totalRead.addAndGet(read);
+                                }
+                            } finally {
+                                inputStream.close();
+                            }
+                        } catch (Exception e) {
+                            readerException = e;
+                            e.printStackTrace();
+                        } finally {
+                            LOG.info(totalRead + " total bytes read.");
+                        }
+                    }
+                });
+
+                final Thread writerThread = new Thread(new Runnable() {
+                    private final Random random = new Random();
+
+                    public void run() {
+                        totalWritten.set(0);
+                        int count = MESSAGE_COUNT;
+                        try {
+                            final OutputStream outputStream = connection.createOutputStream(destination);
+                            try {
+                                final byte[] buf = new byte[BUFFER_SIZE];
+                                random.nextBytes(buf);
+                                while (count > 0 && !stopThreads.get()) {
+                                    outputStream.write(buf);
+                                    totalWritten.addAndGet(buf.length);
+                                    count--;
+                                }
+                            } finally {
+                                outputStream.close();
+                            }
+                        } catch (Exception e) {
+                            writerException = e;
+                            e.printStackTrace();
+                        } finally {
+                            LOG.info(totalWritten + " total bytes written.");
+                        }
+                    }
+                });
+
+                readerThread.start();
+                writerThread.start();
+
+                // Wait till reader is has finished receiving all the messages
+                // or he has stopped
+                // receiving messages.
+                Thread.sleep(1000);
```

Comment:
Instead of a thread.sleep(1000) could you use a countdown latch to determine when the reader has finished?